contrib/mesos/pkg/executor/service/service.go

```diff
@@ -212,7 +215,90 @@ func (s *KubeletExecutorServer) Run(hks hyperkube.Interface, _ []string) error {
 		log.Warningf("Unknown Docker exec handler %q; defaulting to native", s.DockerExecHandlerName)
 		dockerExecHandler = &dockertools.NativeExecHandler{}
 	}
+	dockerClient := dockertools.ConnectToDockerOrDie(s.DockerEndpoint)
+
+	//TODO(jdef) either configure Watch here with something useful, or else
+	// get rid of it from executor.Config
+	kubeletFinished := make(chan struct{})
+	execUpdates := make(chan kubelet.PodUpdate, 1)
+	exec := executor.New(executor.Config{
+		Updates:         execUpdates,
+		APIClient:       apiclient,
+		Docker:          dockerClient,
+		SuicideTimeout:  s.SuicideTimeout,
+		KubeletFinished: kubeletFinished,
+		ShutdownAlert: func() {
+			if shutdownCloser != nil {
+				if e := shutdownCloser.Close(); e != nil {
+					log.Warningf("failed to signal shutdown to external watcher: %v", e)
+				}
+			}
+		},
+		ExitFunc: os.Exit,
+		PodStatusFunc: func(pod *api.Pod) (*api.PodStatus, error) {
+			if s.klet == nil {
```

Comment:
Good spot. We could add a lock. But maybe we have to think about how to do this when executor and kubelet processes are split. Maybe we can also use directly the apiserver client. No need to call into the kubelet.